File: src/main/java/in/dragonbra/javasteam/types/BitVector64.java
package in.dragonbra.javasteam.types;

public class BitVector64 {

    private Long data;

    public BitVector64() {
    }

    public BitVector64(long value) {
        data = value;
    }

    public Long getData() {
        return data;
    }

    public void setData(Long data) {
        this.data = data;
    }

    public long getMask(short bitOffset, long valueMask) {
        return data >> bitOffset & valueMask;
    }

    public void setMask(short bitOffset, long valueMask, long value) {
        data = (data & ~(valueMask << bitOffset)) | ((value & valueMask) << bitOffset);
    }
}

File: src/main/java/in/dragonbra/javasteam/types/BitVector64.kt
package `in`.dragonbra.javasteam.types

class BitVector64(var data: Long) {

    fun getMask(bitOffset: Int, valueMask: Long): Long = (data shr bitOffset) and valueMask

    fun setMask(bitOffset: Int, valueMask: Long, value: Long) {
        data = (data and (valueMask shl bitOffset).inv()) or ((value and valueMask) shl bitOffset)
    }
}
